### PR TITLE
chore: add info in max active sessions error

### PR DESCRIPTION
**Pull request by flaneur2020**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- this PR adds infomation on the current active session and limit value, may make diagnose session problem easier
- also add a gauge metric about how many active sessions in this process, for active sessions, a gauge value is easier to use than counters.

### CODE REVIEW

File: src/query/service/src/sessions/session_mgr.rs [hunk 1]
```diff
@@ -129,34 +123,31 @@ impl SessionManager {
...
+        if !matches!(typ, SessionType::FlightRPC) {
+            sessions.insert(session.get_id(), Arc::downgrade(&session));
+        }
 
-            Ok(session)
-        } else {
-            Err(ErrorCode::TooManyUserConnections(
-                "The current accept connection has exceeded max_active_sessions config",
-            ))
+        if let SessionType::MySQL = session_typ {
```
Comment (BohuTANG) [inline src/query/service/src/sessions/session_mgr.rs:144]: The  HTTP handle session need check or not?

Comment (flaneur2020) [reply src/query/service/src/sessions/session_mgr.rs:144]: the check logic is still same as before, the modification footprint here is caused by this block of code are all inside a `if else`, this change reduced this `if else`.

Comment (BohuTANG) [reply src/query/service/src/sessions/session_mgr.rs:144]: This question NOT releated to this patch, do you need to check http handler max session?

Comment (BohuTANG) [reply src/query/service/src/sessions/session_mgr.rs:144]: This maybe need to be addressed in another issue, let mark resolved.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
test servers::mysql::mysql_handler::test_rejected_session_with_parallel ... FAILED
  2022-11-24T10:43:33.063204Z ERROR common_tracing::panic_hook: panicked at 'assertion failed: `(left == right)`
  left: `"Reject connection, cause: Server error: `ERROR HY000 (1815): Current sessions (1) has exceeded the max_active_sessions limit (1)'"`,
 right: `"Reject connection, cause: Server error: `ERROR HY000 (1815): The current accept connection has exceeded max_active_sessions config'"`', src/query/service/tests/it/servers/mysql/mysql_handler.rs:71:17, backtrace: Backtrace [{ fn: "common_tracing::panic_hook::log_panic" }, { fn: "common_tracing::panic_hook::set_panic_hook::{{closure}}" }, { fn: "<alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2032 }, { fn: "std::panicking::rust_panic_with_hook", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 692 }, { fn: "std::panicking::begin_panic_handler::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 579 }, { fn: "std::sys_common::backtrace::__rust_end_short_backtrace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys_common/backtrace.rs", line: 137 }, { fn: "rust_begin_unwind", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 575 }, { fn: "core::panicking::panic_fmt", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panicking.rs", line: 65 }, { fn: "core::panicking::assert_failed_inner", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panicking.rs" }, { fn: "core::panicking::assert_failed" }, { fn: "it::servers::mysql::mysql_handler::test_rejected_session_with_sequence::{{closure}}" }, { fn: "<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll" }, { fn: "<core::pin::Pin<P> as core::future::future::Future>::poll" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::{{closure}}" }, { fn: "tokio::coop::with_budget::{{closure}}" }, { fn: "std::thread::local::LocalKey<T>::try_with" }, { fn: "std::thread::local::LocalKey<T>::with" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}" }, { fn: "tokio::runtime::scheduler::current_thread::Context::enter" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}" }, { fn: "tokio::macros::scoped_tls::ScopedKey<T>::set" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::enter" }, { fn: "tokio::runtime::scheduler::current_thread::CoreGuard::block_on" }, { fn: "tokio::runtime::scheduler::current_thread::CurrentThread::block_on" }, { fn: "tokio::runtime::Runtime::block_on" }, { fn: "it::servers::mysql::mysql_handler::test_rejected_session_with_sequence" }, { fn: "it::servers::mysql::mysql_handler::test_rejected_session_with_sequence::{{closure}}" }, { fn: "core::ops::function::FnOnce::call_once" }, { fn: "core::ops::function::FnOnce::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/ops/function.rs", line: 510 }, { fn: "test::__rust_begin_short_backtrace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/test/src/lib.rs", line: 637 }, { fn: "test::run_test::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/test/src/lib.rs", line: 628 }, { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/ops/function.rs", line: 510 }, { fn: "<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2000 }, { fn: "<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panic/unwind_safe.rs", line: 271 }, { fn: "std::panicking::try::do_call", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 483 }, { fn: "std::panicking::try", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 447 }, { fn: "std::panic::catch_unwind", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panic.rs", line: 137 }, { fn: "test::run_test_in_process", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/test/src/lib.rs", line: 660 }, { fn: "test::run_test::run_test_inner::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/test/src/lib.rs", line: 554 }, { fn: "test::run_test::run_test_inner::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/test/src/lib.rs", line: 581 }, { fn: "std::sys_common::backtrace::__rust_begin_short_backtrace", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys_common/backtrace.rs", line: 121 }, { fn: "std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/thread/mod.rs", line: 551 }, { fn: "<core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/panic/unwind_safe.rs", line: 271 }, { fn: "std::panicking::try::do_call", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 483 }, { fn: "std::panicking::try", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panicking.rs", line: 447 }, { fn: "std::panic::catch_unwind", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/panic.rs", line: 137 }, { fn: "std::thread::Builder::spawn_unchecked_::{{closure}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/thread/mod.rs", line: 550 }, { fn: "core::ops::function::FnOnce::call_once{{vtable.shim}}", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/core/src/ops/function.rs", line: 510 }, { fn: "<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2000 }, { fn: "<alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/alloc/src/boxed.rs", line: 2000 }, { fn: "std::sys::unix::thread::Thread::new::thread_start", file: "/rustc/e631891f7ad40eac3ef58ec3c2b57ecd81e40615/library/std/src/sys/unix/thread.rs", line: 108 }, { fn: "start_thread" }, { fn: "clone" }], panic.file: "src/query/service/tests/it/servers/mysql/mysql_handler.rs", panic.line: 71, panic.column: 17
```

File: src/query/service/src/sessions/session_mgr.rs (hunk 1 above)
Comment (flaneur2020) [reply src/query/service/src/sessions/session_mgr.rs:144]: > This question NOT releated to this patch, do we need to check http handler max session?

I guess the sessions in HTTP / clickhouse can be taken unified in the self.active_sessions check.

but I am curious about why does mysql do the special check? it have both checks about `mysql_conn_map.len()` and `self.active_sessions`

Comment (BohuTANG) [reply src/query/service/src/sessions/session_mgr.rs:144]: Yes, me too. cc @zhang2014

**Pull request merged**